packages/@lwc/engine-core/src/framework/profiler.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { startMeasure, endMeasure, MeasurementPhase } from './performance-timing';
+import { VM } from './vm';
+
+function noop(_opId: number, _phase: number, _cmpName: string, _vm_idx: number) {}
+
+let logOperation = noop;
+
+export enum OperationId {
+    constructor = 0,
+    render = 1,
+    patch = 2,
+    connectedCallback = 3,
+    renderedCallback = 4,
+    disconnectedCallback = 5,
+    errorCallback = 6,
+}
+
+enum Phase {
+    Start = 0,
+    Stop = 1,
+}
+
+const opIdToMeasurementPhaseMappingArray: MeasurementPhase[] = [
+    'constructor',
+    'render',
+    'patch',
+    'connectedCallback',
+    'renderedCallback',
+    'disconnectedCallback',
+    'errorCallback',
+];
+
+let profilerEnabled = false;
+let logMarks = false;
+let bufferLogging = false;
+
+if (process.env.NODE_ENV !== 'production') {
+    profilerEnabled = true;
+    logMarks = true;
+    bufferLogging = false;
+}
+
+const profilerStateCallbacks: ((arg0: boolean) => void)[] = [];
+
+function trackProfilerState(callback: (arg0: boolean) => void) {
+    callback(profilerEnabled);
+    profilerStateCallbacks.push(callback);
+}
+
+function logOperationStart(opId: OperationId, vm: VM) {
+    if (logMarks) {
+        startMeasure(opIdToMeasurementPhaseMappingArray[opId], vm);
+    }
+    if (bufferLogging) {
+        logOperation(opId, Phase.Start, vm.tagName, vm.idx);
```

Comment:
@caridy switched to using `vm.tagName` directly as it's done in https://github.com/salesforce/lwc/blob/master/packages/@lwc/engine-core/src/shared/format.ts#L12

I don't want the concatenated string, just the raw tag name.